src/blog/0.3/generics-hash-codes-madness.vmd

```diff
@@ -0,0 +1,256 @@
+---
+title: Hash Codes, Non-Determinism, and Other Eldritch Horrors
+author: Evan Ovadia
+date: Nov 6, 2022
+realm: blog
+path: blog/generics-hash-codes-madness
+layout: annotated
+namespace: c-blog m-annotated
+---
+
+
+Until recent events, I've firmly believed the most terrifying thing known to mankind is *the kraken.*
+
+The kraken is a colossal eldritch sea horror which resembles a giant squid. It's known to crush entire ships.
+
+They say that 60% of all ships lost at sea fell victim to krakens, probably. [# I probably didn't make this up.]
+
+<center><img src="/images/kraken.jpg" width="100%" style="max-width: 400px"/></center>
+
+If someone says they aren't scared of krakens, they've obviously never met one.
+
+
+However, something else has started appearing in my kraken nightmares: hash codes. Not all hash codes, but a specific kind of hash code that's more insidious than colossal eldritch sea horrors.
+
+
+This is my tale, where I discovered a terrifying truth underlying our world, and the reason why krakens and hash codes would appear in the same nightmare.
+
+
+
+# A Worthy Goal
+
+
+For the last several months, I've been working on Vale's support for [regions](blog/zero-cost-memory-safety-regions-part-1-immutable-borrowing), which involves adding a sort of "opt-in borrow checker" to the language. Since it's opt-in and not [upwardly infectious](https://www.reddit.com/r/ProgrammingLanguages/comments/vofiyv/thoughts_on_infectious_systems_asyncawait_and_pure/), it should pair well with our [generational references](https://verdagon.dev/blog/generational-references) to give us speed and safety while still keeping the language easy to use. As someone who grew up with C++ and Rust, this makes me _very_ excited.
+
+
+This meant adding more functionality to our type system and function resolution, and transitioning from [templates](https://www.tutorialspoint.com/cplusplus/cpp_templates.htm) to full [generics](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/differences-between-cpp-templates-and-csharp-generics). [# This is because part of regions involves monomorphizing any read-only region into both an immutable and mutable region.]
```

Comment:
> This is because part of regions involves monomorphizing any read-only region into both an immutable and mutable region

Both sentences here start with 'This' which is awkward. I'd go with something like "I realized that read only regions need to me mono..."